test/Dialect/Arc/arc-canonicalizer.mlir

```diff
@@ -491,6 +491,48 @@ hw.module @Repeated_input(in %b: i8, in %e: i8, in %h: i8, in %k: i8, in %en: i1
 // CHECK-NEXT:    hw.output
 // CHECK-NEXT:  }
 
+arc.define @Queue_24_arc(%arg0: i1, %arg1: i1, %arg2: i1, %arg3: i1) -> i124 {
+  %false = hw.constant false
+  %c0_i73 = hw.constant 0 : i73
+  %c0_i46 = hw.constant 0 : i46
+  %0 = comb.concat %c0_i73, %arg0, %arg1, %false, %arg2, %arg3, %c0_i46 : i73, i1, i1, i1, i1, i1, i46
+  arc.output %0 : i124
+}
+arc.define @TLXbar_8_arc() -> i1 {
+  %false = hw.constant false
+  arc.output %false : i1
+}
+arc.define @dummy(%arg0: i124, %arg1: i124) -> i124 {
+  %0 = comb.or %arg0, %arg1 : i124
+  arc.output %0 : i124
+}
+
+hw.module @Repeated_again(in %clock : !seq.clock, in %0 "" : i1) {
+  %1:2 = arc.vectorize (%clock, %clock) : (!seq.clock, !seq.clock) -> (i1, i1) {
+  ^bb0(%arg0: !seq.clock):
+    %4 = arc.state @TLXbar_8_arc() clock %arg0 latency 1 : () -> i1
+    arc.vectorize.return %4 : i1
+  }
```

Comment:
Is this `arc.vectorize` necessary? Isn't just the other one triggering this issue?

Reply:
It isn't, I just needed multiple inputs to handle the shuffles. I will handle it with another input.

Reply:
how about this? If there's a way to return the result from the module, that would be great as we will we git rid of the arc.

```
arc.define @dummy(%arg0: i1, %arg1: i1) -> i1 {
  %0 = comb.or %arg0, %arg1 : i1
  arc.output %0 : i1
}

hw.module @Repeat_agian(in %clock : !seq.clock, in %0 : i1, in %in1: i1, in %in2: i1) {
  %2:2 = arc.vectorize (%in1, %in2), (%in1, %in2), (%0, %0), (%0, %0) : (i1, i1, i1, i1, i1, i1, i1, i1) -> (i1, i1) {
  ^bb0(%arg0: i1, %arg1: i1, %arg2: i1, %arg3: i1):
    %4 = comb.or %arg0, %arg1, %arg2, %arg3 : i1
    arc.vectorize.return %4 : i1
  }
  %3 = arc.state @dummy(%2#0, %3) clock %clock latency 1 : (i1, i1) -> i1
  hw.output
}
```